internal/sysfs/osfile.go

```diff
@@ -98,9 +98,38 @@ func (f *osFile) reopen() (errno experimentalsys.Errno) {
 	// Clear any create flag, as we are re-opening, not re-creating.
 	f.flag &= ^experimentalsys.O_CREAT
 
+	var (
+		isDir  bool
+		offset int64
+		err    error
+	)
+
+	isDir, errno = f.IsDir()
+	if errno != 0 {
+		return errno
+	}
+
+	if !isDir {
+		offset, err = f.file.Seek(0, 1)
```

Comment:
Maybe too late for a nit, but please constants for the whence: `io.SeekStart`, etc…

Reply:
Good point, let's fix it in a follow up PR